RELEASE_PROCEDURE.md

```diff
@@ -0,0 +1,39 @@
+# Release procedure
+
+## Preparation
+
+- Consolidate the [release notes](docs/root/version_history.md).
+  - Make sure breaking changes are explicitly called out.
+    Ideally these have been tracked on the go but it does not hurt to ask around and double
+    check. Running the python integration tests of the previous release with the latest binaries could serve as a double check for this.
+  - Ensure the release notes are complete. Ideally these have been tracked on the go.
+- Based on our [semantic versioning](https://semver.org/spec/v2.0.0.html) strategy, our release numbers look like `MAJOR.MINOR.PATCH`. Determine the release type to come up with what the next version should look like:
+  - Breaking changes increment `MAJOR`, and reset `MINOR.PATCH` to `0.0`
+  - Added functionality which is backwards compatible increments `MINOR`, and resets `PATCH` to `0`
+  - Bug fixes increment `PATCH`
+
+## Release steps
+
+1. Speculatively bump the version in [version.h](include/nighthawk/common/version.h) to the version you determined earlier. This may result in version gaps if a release attempt fails, but avoids having to freeze merges to master and/or having to work with release branches. In short it helps keeping the release procedure lean and mean and eliminates the need for blocking others while this procedure is in-flight.
```

Comment:
version.h is coming in a separate PR that may be delayed. Do we want to pull it in here?

Reply:
Good point, added a pre-requisite to land this in the [PR description](https://github.com/envoyproxy/nighthawk/pull/232#issue-353280027)

Reply:
Also tagged this is blocked for that reason